package/MDAnalysis/analysis/density.py

```diff
@@ -508,15 +516,22 @@ def current_coordinates():
     grid *= 0.0
     h = grid.copy()
 
+    pm = ProgressMeter(u.trajectory.n_frames, interval=1,
+                       format="Histogramming %(n_atoms)6d atoms in frame "
+                       "%(step)5d/%(numsteps)d  [%(percentage)5.1f%%]\r")
     for ts in u.trajectory:
-        print("Histograming %6d atoms in frame %5d/%d  [%5.1f%%]\r" % \
-              (len(coord), ts.frame, u.trajectory.n_frames, 100.0 * ts.frame / u.trajectory.n_frames),)
-        coord = current_coordinates()
+        if update_selection:
+           group = u.select_atoms(atomselection)
+           coord=group.positions
+        else:
+           coord = current_coordinates()
+
+        pm.echo(ts.frame, n_atoms=len(coord))
```

Comment:
There's the new `quiet` keyword, which can be used to mute `ProgressMeter`, it's in 5958fa92ac0d6067d2792c34c83ad77e02a3d6de.